bgpd/bgp_routemap.c

```diff
@@ -1382,14 +1382,22 @@ static route_map_result_t route_match_interface(void *rule,
 {
 	struct interface *ifp;
 	struct bgp_path_info *path;
+	struct bgp_node *rn;
+	struct bgp_table *table;
 
 	if (type == RMAP_BGP) {
 		path = object;
 
 		if (!path || !path->attr)
 			return RMAP_NOMATCH;
 
-		ifp = if_lookup_by_name_all_vrf((char *)rule);
+		/* get vrf_id of current bgp instance */
+		rn = path->net;
+		assert(rn && bgp_node_table(rn));
```

Comment:
This makes sense to me, but since #3846 wasn't fixed yet (AFAIK), I'm afraid this change can cause bgpd to crash more often :disappointed:

Reply:
I will remove the assert, and adapt so that route map will not match if bgp is not reachable